Jenkinsfile

```diff
@@ -60,60 +88,60 @@ timestamps() {
 					sh 'npx retire --exitwith 0'
 
 					step([$class: 'WarningsPublisher', canComputeNew: false, canResolveRelativePaths: false, consoleParsers: [[parserName: 'Node Security Project Vulnerabilities'], [parserName: 'RetireJS']], defaultEncoding: '', excludePattern: '', healthy: '', includePattern: '', messagesPattern: '', unHealthy: ''])
-				} // stage
+				} // nodejs
+			} // stage
+		} // node
 
-				stage('Publish') {
+		node('(osx || linux) && git && npm-publish') {
+			stage('Publish') {
+				checkout scm
+				nodejs(nodeJSInstallationName: "node ${nodeVersion}") {
 					if (!isPR) {
 						try {
 							// Publish
 							sh 'npm publish'
+
 							// Tag git
 							pushGitTag(name: packageVersion, message: "See ${env.BUILD_URL} for more information.", force: true)
+
 							// Trigger appc-cli job
-							build job: '../appc-cli/master', wait: false
-							// Set published to true so we can update tickets
-							published = true
+							build job: '../appc-cli/master', wait: false, parameters: [ 
+								[$class: 'StringParameterValue', name: 'packageName', value: 'alloy' ],
+								[$class: 'StringParameterValue', name: 'packageVersion', value: packageVersion ],
+							]
+							// Update tickets
+							updateJIRA('ALOY', "Alloy ${packageVersion}", scm)
 						} catch (e) {
 							// Don't thow the errors as we don't want a failed publish due to the version not being bumped
 							// being classed as a failure on the build
 						}
-					}
-				}
-
-				stage('JIRA') {
-					if (published) {
-						def versionName = "Alloy ${packageVersion}"
-						def projectKey = 'ALOY'
-						def issueKeys = jiraIssueSelector(issueSelector: [$class: 'DefaultIssueSelector'])
-
-						// Comment on the affected tickets with build info
-						step([
-							$class: 'hudson.plugins.jira.JiraIssueUpdater',
-							issueSelector: [$class: 'hudson.plugins.jira.selector.DefaultIssueSelector'],
-							scm: scm
-						])
-
-						// Create the version we need if it doesn't exist...
-						step([
-							$class: 'hudson.plugins.jira.JiraVersionCreatorBuilder',
-							jiraVersion: versionName,
-							jiraProjectKey: projectKey
-						])
-
-						// Should append the new version to the ticket's fixVersion field
-						def fixVersion = [name: versionName]
-						for (i = 0; i < issueKeys.size(); i++) {
-							def result = jiraGetIssue(idOrKey: issueKeys[i])
-							def fixVersions = result.data.fields.fixVersions << fixVersion
-							def testIssue = [fields: [fixVersions: fixVersions]]
-							jiraEditIssue(idOrKey: issueKeys[i], issue: testIssue)
+					} // if
+				} // nodejs
+			} // stage
+		} // node
+	} finally {
+		if (runDanger) {
+			stage('Danger') {
+				node('osx || linux') {
+					nodejs(nodeJSInstallationName: "node ${nodeVersion}") {
+						checkout scm
```

Comment:
again, if you wanted to be stingy about network, you could just unstash package.json, package-lock.json and dangerfile.js here I think.

Reply:
Left this as it due to the eslint step requiring the js files